cli/azd/cmd/root.go

```diff
@@ -30,25 +31,37 @@ func NewRootCmd(staticHelp bool, middlewareChain []*actions.MiddlewareRegistrati
 	opts := &internal.GlobalCommandOptions{GenerateStaticHelp: staticHelp}
 	opts.EnableTelemetry = telemetry.IsTelemetryEnabled()
 
-	rootCmd := &cobra.Command{
-		Use:   "azd",
-		Short: "Azure Developer CLI is a command-line interface for developers who build Azure solutions.",
-		//nolint:lll
-		Long: `Azure Developer CLI is a command-line interface for developers who build Azure solutions.
+	productName := "Azure Developer CLI"
+	if opts.GenerateStaticHelp {
```

Comment:
For these types of cases, is there a different way we can restructure the doc generation instead of using the static help functionality? (As it doesn't actually pertain to dynamic data)

I can also understand why we might just do this pragmatically. Just thinking about any formalization we can make as to the type of modifications for doc gen, and have easier ways to make these types of changes.

Reply:
Agreed. I think we can revisit this as part of the docs/UX work that we're doing.

In this case we might want to have a rule processing the ref docs that says "when you see 'azd' separated by whitespace or at the start or end of a line in a line outside of a codefence , wrap it in backticks".

Reply:
https://github.com/Azure/azure-dev/issues/1362